emulator/src/tensil/tools/emulator/Main.scala

```diff
@@ -0,0 +1,223 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright © 2019-2022 Tensil AI Company */
+
+package tensil.tools.emulator
+
+import java.io.{
+  File,
+  FileInputStream,
+  DataInputStream,
+  DataOutputStream,
+  FileOutputStream,
+  ByteArrayInputStream,
+  ByteArrayOutputStream
+}
+
+import scala.reflect.ClassTag
+import scala.collection.mutable
+
+import tensil.{
+  Architecture,
+  ArchitectureDataType,
+  ArchitectureDataTypeWithBase,
+  InstructionLayout,
+  FloatAsIfIntegralWithMAC,
+  NumericWithMAC
+}
+import tensil.tools.ArchitectureDataTypeUtil
+import tensil.tools.model.Model
+
+case class Args(
+    modelFile: File = new File("."),
+    inputFiles: Seq[File] = Nil,
+    outputFiles: Seq[File] = Nil,
+    numberOfRuns: Int = 1,
+)
+
+object Main extends App {
+  val argParser = new scopt.OptionParser[Args]("compile") {
+    help("help").text("Prints this usage text")
+
+    opt[File]('m', "model")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(modelFile = x))
+      .text("Tensil model (.tmodel) file")
+
+    opt[Seq[File]]('i', "inputs")
+      .required()
+      .valueName("<file>, ...")
+      .action((x, c) => c.copy(inputFiles = x))
+      .text("Input (.csv) files")
+
+    opt[Seq[File]]('o', "outputs")
```

Comment:
Should it be possible to run without outputs specified so you can just see what the results would be? I'm imagining it might be useful for someone just wanting to do a quick sanity check

Reply:
Yes, good point. Will add.

Reply:
Done. Example of the output for ResNet20:

```
---------------------------------------------------------------------------------------------------------
OUTPUT Identity:0, RUN 0
---------------------------------------------------------------------------------------------------------
00000000       1.6992    -16.7188    -10.2227     -0.5859     -1.9258     -7.0547     -8.1211    -13.7266
00000001      -4.7383    -12.3672      0.0000      0.0000      0.0000      0.0000      0.0000      0.0000
---------------------------------------------------------------------------------------------------------
```